### ISSUE TITLE
Implement OTP for Two Factor Authentication

### ISSUE THREAD

**Issue by dacrome** [osiam/osiam#78]
Moved from https://github.com/osiam/auth-server/issues/41:

@wallner:

> Provide the ability to use two factor authentication with OSIAM. It would be a good feature to provide an implementation of OTP as specified in [RFC 6238](https://tools.ietf.org/html/rfc6238). For a start it might to be a good idea to take a look at [GoogleAuth](https://github.com/wstrange/GoogleAuth) and see how far we can get with that.

@tkrille:

> Is there a `spring-security-...` component for that?

@thomasdarimont:

> Hi, there are several impls / examples but nothing official AFAIK.
> - [spring-security-otp](https://github.com/upcrob/spring-security-otp) 
> - [grails-spring-security-otp](https://github.com/osoco/grails-spring-security-otp)
> - [spring-security-otp-sample](https://github.com/osoco/spring-security-otp-sample)
> - [two-factor-authentication-with-spring](http://altfatterz.blogspot.de/2014/02/two-factor-authentication-with-spring.html)
> - [authentication-using-one-time-password-otp-technique-part-1](http://vitalflux.com/authentication-using-one-time-password-otp-technique-part-1/)
> - [spring-security-otp-plugin-for-grails](https://prezi.com/mbqtxenlpsio/spring-security-otp-plugin-for-grails/)

**Comment by cornelinux** [comment]
Why not stick to old ideas and integrate the ready made solutions https://github.com/privacyidea/privacyidea. Originally I implemented the SCIM user connector of privacyIDEA to work with osiam...

You can simply use the REST API to get OTP authentication for HOTP, TOTP, Yubikey, EMail, SMS... and many more
--end of marketing--

**Comment by wallner** [comment]
@cornelinux You are right of course. We need to take a look at privacyIDEA. As it seems OTP is is not going to be part of OSIAM 3.0.
